trade-finance-manager-api/api-tests/v1/amendments/create-amendment-tfm-object-on-amendment-completion.api-test.js

```diff
@@ -52,89 +63,177 @@ describe('update amendment-tfm on amendment completion', () => {
+    });
+
+    it('createAmendmentTFMObject() - should create tfm object when getLatestCompletedAmendmentValue only', async () => {
+      externalApis.getLatestCompletedAmendmentDate = jest.fn(() => Promise.resolve({}));
+
+      const result = await amendmentController.createAmendmentTFMObject(mockAmendment.amendmentId, mockAmendment.facilityId);
+
+      const expected = {
+        exposure: {
+          exposure: '4,000.00',
+          timestamp: expect.any(Number),
+          ukefExposureValue: 4000,
+        },
+        value: {
+          currency: CURRENCY.GBP,
+          value: 5000,
+        },
+      };
+
+      expect(result).toEqual(expected);
+    });
+
+    it('createAmendmentTFMObject() - should create tfm object when only getLatestCompletedAmendmentDate', async () => {
+      externalApis.getFacilityExposurePeriod = jest.fn(() => Promise.resolve({ exposurePeriodInMonths: 5 }));
+      externalApis.getLatestCompletedAmendmentValue = jest.fn(() => Promise.resolve({}));
+
+      const result = await amendmentController.createAmendmentTFMObject(mockAmendment.amendmentId, mockAmendment.facilityId);
+
+      const expected = {
+        amendmentExposurePeriodInMonths: 5,
+        coverEndDate: unixTime,
+      };
+
+      expect(result).toEqual(expected);
+    });
   });
 
-  it('createAmendmentTFMObject() - should create tfm object when getLatestCompletedAmendmentValue only', async () => {
-    externalApis.getLatestCompletedAmendmentDate = jest.fn(() => Promise.resolve({}));
-
-    const result = await amendmentController.createAmendmentTFMObject(mockAmendment.amendmentId, mockAmendment.facilityId);
-
-    const expected = {
-      exposure: {
-        exposure: '4,000.00',
-        timestamp: expect.any(Number),
-        ukefExposureValue: 4000,
-      },
-      value: {
-        currency: CURRENCY.GBP,
-        value: 5000,
-      },
-    };
-
-    expect(result).toEqual(expected);
-  });
-
-  it('createAmendmentTFMObject() - should create tfm object when only getLatestCompletedAmendmentDate', async () => {
-    externalApis.getFacilityExposurePeriod = jest.fn(() => Promise.resolve({ exposurePeriodInMonths: 5 }));
-    externalApis.getLatestCompletedAmendmentValue = jest.fn(() => Promise.resolve({}));
-
-    const result = await amendmentController.createAmendmentTFMObject(mockAmendment.amendmentId, mockAmendment.facilityId);
-
-    const expected = {
-      amendmentExposurePeriodInMonths: 5,
-      coverEndDate: unixTime,
-    };
-
-    expect(result).toEqual(expected);
+  describe('TFM Facility end date feature flag enabled', () => {
+    beforeEach(() => {
+      jest.mocked(isTfmFacilityEndDateFeatureFlagEnabled).mockReturnValue(true);
+    });
+
+    it('createAmendmentTFMObject() - should not create tfm object when no getLatestCompletedAmendmentValue, getLatestCompletedAmendmentDate or getLatestFacilityEndDate', async () => {
+      externalApis.getLatestCompletedAmendmentValue = jest.fn(() => Promise.resolve({}));
```

Comment:
Doesn't make a change to implementation, but this (and below) can be written as `jest.fn().mockResolvedValue({})`